### ISSUE TITLE
Add-on Store upgrade should be more graceful: All add-ons become external (user must reinstall all to make them updatable)

### ISSUE THREAD

**Issue by XLTechie** [nvaccess/nvda#15006]
### Steps to reproduce:

* Install a version of NVDA prior to a version containing the store. E.g. alpha-27996,d8396f9b.
* Install multiple add-ons.
* Update to a 2023.2 pre-release.

### Actual behavior:

Upon opening the store, all add-ons show as external.
Out of date add-ons show no updates, even though the store contains newer versions.

### Expected behavior:

There should be an automatic upgrade process of some sort, so that when switching to a store version of NVDA, existing add-ons will be detected, and metadata will be integrated with the store.

To require all users to re-install all of their add-ons, is going to be very off-putting to users.
Especially since some add-ons include accumulated data, which might be compromised by doing this, if the user does it wrong.

#### Alternate proposal:

External add-ons should have update checks run, based on name and version number. Thereby external add-ons with store updates, can be updated.

### NVDA logs, crash dumps and other attachments:

[NVDA log after store upgrade.txt](https://github.com/nvaccess/nvda/files/11753330/NVDA.log.after.store.upgrade.txt)

Note, in the above log, I am unclear how Event Tracker became enabled. It had been disabled prior to the upgrade. However this installation was afflicted by the installer creating metadata files bug, which may have enabled it the last time I ran this copy.

#### NVDA installed/portable/running from source:

Installed

#### NVDA version:

alpha-28430,3920fa58

#### Windows version:

10.0.19041.1

#### Name and version of other software in use when reproducing the issue:

N/A

#### Does the issue still occur after restarting your computer?

Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.

Portable copies demonstrate the same behavior as installed copies.

#### If NVDA add-ons are disabled, is your problem still occurring?

N/A

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

N/A but yes

**Comment by seanbudd** [comment]
>Out of date add-ons show no updates, even though the store contains newer versions.

>External add-ons should have update checks run, based on name and version number. Thereby external add-ons with store updates, can be updated

This should not be the case and the solution you describe should be the case if the add-on ID matches. Could you give an example of an add-on that should have an update but doesn't? Please include the add-on ids and add-on version strings.

**Comment by XLTechie** [comment]
Issue withdrawn. I can not replicate this on a clean setup.
